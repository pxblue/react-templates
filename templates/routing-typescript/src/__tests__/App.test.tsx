import React from 'react';
import { render, screen } from '@testing-library/react';
import { ThemeProvider, StyledEngineProvider } from '@mui/material/styles';
import { HomePage } from '../pages/home';
import { DrawerContext } from '../contexts/drawerContextProvider';
import { blueThemes } from '@brightlayer-ui/react-themes';
import { test, vi } from 'vitest';

test('renders welcome text', () => {
    render(
        <StyledEngineProvider injectFirst>
            <ThemeProvider theme={blueThemes}>
                <DrawerContext.Provider
                    value={{
                        drawerOpen: true,
                        setDrawerOpen: vi.fn(),
                    }}
                >
                    <HomePage />
                </DrawerContext.Provider>
            </ThemeProvider>
        </StyledEngineProvider>
    );
    const bluiText = screen.getByText(/Welcome to Brightlayer/i);
    expect(bluiText).toMatchSnapshot();
});
