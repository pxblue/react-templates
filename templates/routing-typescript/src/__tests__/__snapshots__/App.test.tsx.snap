// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders welcome text 1`] = `
<p
  class="MuiTypography-root MuiTypography-h2 MuiTypography-paragraph css-1fm3dbj-MuiTypography-root"
>
  Welcome to Brightlayer
   
  <span
    class="MuiBox-root css-107kugp"
  >
    UI
  </span>
  .
</p>
`;
